### PR TITLE
Op-Panel changes for new system Bonnell

### PR DESCRIPTION
**Pull request by GiridhariKrishna**
Addition of LCD Device Path, IM Keyword for New System Bonnell support.

### CODE REVIEW

**Comment by geissonator** [comment]
add to approvelist

File: src/panel_app_main.cpp [hunk 1]
```diff
@@ -21,13 +21,17 @@ panel::types::PanelDataMap baseDataMap = {
       panel::constants::rainBaseDbusObj}},
     {panel::constants::everestIM,
      {panel::constants::baseDevPath, panel::constants::devAddr,
-      panel::constants::everBaseDbusObj}}};
+      panel::constants::everBaseDbusObj}},
+    {panel::constants::bonnellIM,
+     {panel::constants::baseDevPath, panel::constants::devAddr,
+      panel::constants::bonelBootFailChipFRU}}};
```
Comment (jinuthomas) [inline src/panel_app_main.cpp:27]: bonelBootFailChipFRU why name change should it not be bonnellBaseDbusObj

**Review by jinuthomas — CHANGES_REQUESTED** [review]
I also see Bonnell missing in static const types::PICFRUPathMap bootFailPIC 

I HAVE NOT COMPLETED THE REVIEW, could you please take a relook at if any changes needed elsewhere or not .

File: include/const.hpp [hunk 2]
```diff
@@ -29,11 +30,14 @@ static constexpr auto everLcdDbusObj =
     "panel1";
 static constexpr auto everBMCObj =
     "/xyz/openbmc_project/inventory/system/chassis/motherboard/bmc";
+static constexpr auto bonelBootFailChipFRU = systemDbusObj;
```
Comment (PriyangaRamasamy) [inline include/const.hpp:33]: bonnell* spellcheck

Comment (PriyangaRamasamy) [reply include/const.hpp:33]: @GiridhariKrishna  As we discussed today, bonnellBootFailChipFRU is systemDbusObj thats right, but the place where you have fed this information is wrong. The boot fail PIC FRU details are in this map https://github.com/ibm-openbmc/ibm-panel/blob/1050/include/const.hpp#L80 . please update this map for bonnell.

and in this baseDataMap https://github.com/ibm-openbmc/ibm-panel/blob/1050/src/panel_app_main.cpp#L12
you need to provide bonnellBaseDbusObj path and not the bootFailPIC fru path.

File: src/panel_app_main.cpp [hunk 3]
```diff
@@ -21,13 +21,17 @@ panel::types::PanelDataMap baseDataMap = {
...
 
 std::string getInputDevicePath(const std::string& imValue)
 {
     if (imValue == panel::constants::rain2s2uIM ||
         imValue == panel::constants::rain2s4uIM ||
-        imValue == panel::constants::rain1s4uIM)
+        imValue == panel::constants::rain1s4uIM ||
+        imValue == panel::constants::bonnellIM)
     {
         return "/dev/input/by-path/platform-1e78a400.i2c-bus-event-joystick";
```
Comment (PriyangaRamasamy) [inline src/panel_app_main.cpp:36]: can you please point me to the place where you confirmed that this is the input device path for bonnell panel.

File: include/const.hpp (hunk 2 above)
Comment (GiridhariKrishnan) [reply include/const.hpp:33]: > bonnell* spellcheck

It's intentional

File: src/panel_app_main.cpp (hunk 3 above)
Comment (jinuthomas) [reply src/panel_app_main.cpp:36]: This will be the same as in Rainier as the i2c for LCD panel is same.

**Review by jinuthomas — CHANGES_REQUESTED** [review]
The configurations are wrong

File: include/const.hpp (hunk 2 above)
Comment (PriyangaRamasamy) [reply include/const.hpp]: that should be ok to give complete word - bonnell with right spelling. its your call.

Comment (PriyangaRamasamy) [reply include/const.hpp]: > and in this baseDataMap https://github.com/ibm-openbmc/ibm-panel/blob/1050/src/panel_app_main.cpp#L12
> you need to provide bonnellBaseDbusObj path and not the bootFailPIC fru path.

can you please address this comment?

File: src/panel_app_main.cpp (hunk 3 above)
Comment (PriyangaRamasamy) [reply src/panel_app_main.cpp:36]: ok

File: include/const.hpp (hunk 2 above)
Comment (GiridhariKrishnan) [reply include/const.hpp]: Yes, initially thought so but confused with systemDbusobj, will change that.

Comment (GiridhariKrishnan) [reply include/const.hpp]: Sorry, changed that already, will update.

**Review by jinuthomas — CHANGES_REQUESTED** [review]
overall looks good , once the nit is done , we can merge this

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**